src/marker.coffee

```diff
@@ -382,8 +382,10 @@ class Marker
     @layer.markerUpdated() if updated and not suppressMarkerLayerUpdateEvents
     updated
 
-  getSnapshot: (range) ->
-    Object.freeze({range, @properties, @reversed, @tailed, @valid, @invalidate, @exclusive})
+  getSnapshot: (range, includeMarker=true) ->
+    snapshot = {range, @properties, @reversed, @tailed, @valid, @invalidate, @exclusive}
+    snapshot.marker = this if includeMarker
```

Comment:
In the case where we include the marker, do you think it's worth exploring the approach of ditching the plain object altogether and using the marker itself as the snapshot? Like, in `MarkerLayer.createSnapshot`, we would 
replace `marker.getSnapshot(...)` with just `marker`.

I'm not sure if it'd be worth it. The reason I suggest it is that for the most part, the snapshot is already just a copy of the marker. The one additional property is `range`. I'm not sure how we'd deal with that... we used to store a property called `rangeWhenDestroyed` on destruction so that you could still query a destroyed marker's range.

Reply:
Yeah, nevermind. I don't really have any ideas for a better way to do this actually.

Reply:
Yeah, it's tough because the marker is mutable. I don't feel like this adds much overhead to the current snapshot. Just one more pointer to a shared object.